app/controllers/users_controller.rb

```diff
@@ -8,5 +8,6 @@ def index
   # based on the provided ID.
   def show
     @user = User.find(params[:id])
+    @posts = @user.posts
   end
 end
```

Comment:
Good job having the correct setup, however, due to some improper set-up in the migrations, there is an error thrown when a post is attempted to be created and it blocks the proper review of this project as the views can not be tested to see if they work as expected and also after users were created, I don't seem to see the users displayed on the root page. I recommend addressing these issues in order to make the creation of resources possible. also consider rubbing the below code piece from Microverse to see if all is fine before requesting another review, I have dropped it below 


HERE IS THE ERROR THROWN

<img width="641" alt="image" src="https://github.com/CaptainBawa/Blog-App/assets/84930355/b090f447-8c25-411e-bf13-e4b475fc6964">


**Here is the code piece to run in the console**

first run `rails c` in the terminal

 ```
first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')

second_user = User.create(name: 'Lilly', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Poland.')

first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post')

Comment.create(post: first_post, author: second_user, text: 'Hi Tom!' )


```

Reply:
Hi @DuaneDave, this code 
### "first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.')"

was working very fine previously but after adding validation to post_counter, you need to create a user this way 

### "first_user = User.create(name: 'Tom', photo: 'https://unsplash.com/photos/F_-0BxGuVvo', bio: 'Teacher from Mexico.', post_counter: 0)"

Reply:
Hi @CaptainBawa the problem occurs when i attempt creating a post

Reply:
hi @DuaneDave 

For post creation, it needs to be like this:

### first_post = Post.create(author: first_user, title: 'Hello', text: 'This is my first post', comments_counter: 0, likes_counter: 0)

These all happened the moment we added validation to post_counter, comments_counter, and likes_counter